static/app/components/profiling/FrameStack/frameStack.tsx

```diff
@@ -23,19 +25,32 @@ function FrameStack(props: FrameStackProps) {
   const {selectedNode} = useFlamegraphProfilesValue();
 
   const [tab, setTab] = useState<'bottom up' | 'call order'>('call order');
+  const [treeType, setTreeType] = useState<'all' | 'application' | 'system'>('all');
   const [recursion, setRecursion] = useState<'collapsed' | null>(null);
 
-  const roots = useMemo(() => {
+  const roots: FlamegraphFrame[] | null = useMemo(() => {
     if (!selectedNode) {
       return null;
     }
 
+    const skipFunction =
+      treeType === 'application'
+        ? (f: FlamegraphFrame): boolean => !f.frame.is_application
```

Comment:
Yes, it can infer for this case, but the return type has to be boolean, that's why I added the type. The better way to type it here would be skipFunction: typeof skipfn or something so it's always just kept in sync witht he filter tree implementation

Reply:
You could declare type on `skipFunction` like
```typescript
const skipFunction: (f: FlamegraphFrame) => boolean = ...
```